app/lib/exchange-rate/money.ts

```diff
@@ -0,0 +1,480 @@
+    this.assertSameCurrency(money);
+    const amountInMinUnit = money.amount(this._data.amountUnit.name);
+    return this._data.amount.gte(amountInMinUnit);
+  };
+
+  lessThan = (money: Money): boolean => {
+    this.assertSameCurrency(money);
+    const amountInMinUnit = money.amount(this._data.amountUnit.name);
+    return this._data.amount.lt(amountInMinUnit);
+  };
+
+  lessThanOrEqual = (money: Money): boolean => {
+    this.assertSameCurrency(money);
+    const amountInMinUnit = money.amount(this._data.amountUnit.name);
+    return this._data.amount.lte(amountInMinUnit);
+  };
+
+  isZero = (): boolean => {
+    return this._data.amount.eq(0);
+  };
+
+  isPositive = (): boolean => {
+    return this._data.amount.gt(0);
+  };
+
+  isNegative = (): boolean => {
+    return this._data.amount.lt(0);
+  };
+
+  /**
+   * Converts money to a regular javascript number for the requested currency unit.
+   * If no unit is provided, base/default unit is used (bitcoin for BTC, dollar for USD, etc.)
+   * Throws an error if it's not possible to keep full precision.
+   */
+  toNumber = (unit?: string): number => {
+    // Don't use big.js toNumber because it sometimes returns -0.
+    const str = this.toString(unit);
+    const num = Number(str);
+
+    if (
+      str !==
+      this.merge({
+        amount: num,
+        unit: unit ?? this.currencyData.baseUnit,
+      }).toString(unit)
+    ) {
+      throw new Error('Converting to number was imprecise');
+    }
+
+    return num;
+  };
+
+  /**
+   * Returns the money value as a stringified number without a currency.
+   *
+   * If you want to include the currency as well use `toLocaleString` instead.
+   *
+   * @param unit Specifies the currency unit to return. If not provided the base/default unit is used (bitcoin for BTC,
+   * dollar for USD, etc.)
+   */
+  toString = (unit?: string): string => {
+    const currencyUnit = this.getCurrencyUnit(unit);
+    const amount = this.amount(currencyUnit.name);
+    return amount.toFixed(currencyUnit.decimals);
+  };
+
+  /**
+   * Returns the money value as a localized stringified number.
+   * @param locale Locale to use to format the number. If not provided the locale of the machine will be used.
+   * @param unit Specifies the currency unit to return. If not provided the base/default unit is used (bitcoin for BTC,
+   * dollar for USD, etc.)
+   * @param showCurrency Controls if the currency symbol should be included in the string. True by default
+   */
+  toLocaleString = ({
+    locale,
+    unit,
+    showCurrency = true,
+  }: {
+    locale?: string;
+    unit?: string;
+    showCurrency?: boolean;
+  } = {}): string => {
+    const currencyUnit = this.getCurrencyUnit(unit);
+    const options: FormatOptions = { locale };
+    if (showCurrency) {
+      options.currency = this.currency;
+    }
+    return currencyUnit.format(this.toNumber(unit), options);
+  };
+
+  toJSON = () => {
+    return {
+      amount: this.toNumber(),
```

Comment:
should we print the number in base/default unit here (like atm) or maybe change it to print it in initial unit?

with the current version of code if you do:

```
const x = Money.create({ amount: 1000, currency: 'BTC', unit: 'satoshi' })
```

and then you do:
```
console.log(`x.getAmount(): `, x.getAmount())
console.log(`x.toJSON(): `, x.toJSON())
```

you would get this printed:
```
x.getAmount(): 0.00001000
x.toJSON(): { amount: 0.00001000, currency: 'BTC' }
```

I wonder if that is weird or we should change outputs to respect the initial unit so it would print:
```
x.getAmount(): 1000
x.toJSON(): { amount: 1000, currency: 'BTC', unit: 'satoshi' }
```

what do you think?

Reply:
if I understand  correctly, there are two unit values. One (`initialUnit`) is the unit of the amount used to create the `Money`. The other (`amountUnit`) is just the smallest unit for this currency and that is the unit that the value is stored under (ie for BTC `amountUnit` will always be millisatoshi). It seems like we are missing a defaultUnit.

In the example you show here I think that having the `initialUnit` be respected is the way to go. However, I wonder if it would be better to pick the default unit when `Money` is constructed so that we can create a satoshi `Money` out of bitcoin. Or a `Money` with `cent` as the default unit from a dollar amount.

Then the construction would be something like:
```js
const x = Money.create({ amount: 1, currency: 'USD', unit: 'cent', fromUnit: 'dollar' })
```

And then we would get:
```js
x.getAmount(): 100 // 100 cents
x.toJSON(): { amount: 100, currency: 'USD', unit: 'cent' }
```

Also, I assume you mean `x.amount()`? I don't see a `getAmount` method.

Reply:
> It seems like we are missing a defaultUnit

what do you mean?

> Also, I assume you mean x.amount()? I don't see a getAmount method.

yes. typo

> However, I wonder if it would be better to pick the default unit when Money is constructed so that we can create a satoshi Money out of bitcoin. Or a Money with cent as the default unit from a dollar amount.
>
> Then the construction would be something like:
>
> const x = Money.create({ amount: 1, currency: 'USD', unit: 'cent', fromUnit: 'dollar' })

Hm, not sure. To many options seem like it could make the api confusing. I'd rather avoid that

Reply:
If you think that having the initial value respected would be better than using the base/default I would maybe only change that for now

Reply:
I guess one drawback of this would be that with inital value you would always have to think about what you passed initially while with current approach you can just know if you didn't provide anything then it is a base value

Reply:
maybe we can even keep as is for now and see how it works in practice